docs/docs/reference/02-yaml-syntax.md

```diff
@@ -0,0 +1,638 @@
+---
+sidebar_position: 1
+---
+
+# YAML application syntax
+
+The following sections define the YAML syntax used to describe an AICA application.
+
+## Overview
+
+An application description contains the following top-level elements.
+
+```yaml
+on_start:
+  ...
+
+hardware_interfaces:
+  ...
+
+conditions:
+  ...
+
+components:
+  ...
+```
+
+## On Start
+
+The `on_start` keyword is reserved as a special event trigger when the application is launched.
+List the [events](#events) to trigger on startup (for example, to load components).
+
+```yaml
+on_start:
+  load:
+    - component_a
+    - component_b
+```
+
+## Hardware Interfaces
+
+Hardware interfaces describe the connected robots and their corresponding controllers.
+
+```yaml
+hardware_interfaces:
+  robot_a:
+    urdf: ...
+    rate: ...
```

Comment:
Yes, it's a fair point. The name `period` was held over from the modulo `Cell` days because it was the input to the wall timer function (a period for the timer to wait). Conversely the `rate` term was adopted from the ros2_control stack, where control engineers like to think about frequencies. 

I don't know which one of period or frequency/rate is more intuitive, but mixing the two in one syntax is not ideal. A `period` for periodic behavior is quite nice conceptually, but I would also not be against components having a rate instead. Only problem is that it would be a breaking change unless we support both parameters, and in that case we would need to decide which one to prioritize if both are set.

You could open a corresponding issue in modulo if you want.

Reply:
For me, a frequency is more intuitive because you can ask yourself how many times a second should it execute which in my opinion is easier for the brain to imagine that saying how long on period should be

Reply:
Opened it :+1: